javaslang/src/main/java/javaslang/collection/BitMappedTrie.java

```diff
@@ -345,6 +345,18 @@ private int filter(Predicate<? super T> predicate, Object results, int index, T
         return index;
     }
 
+    Object[] toArray() {
+        final Object[] results = new Object[length];
+        visit((index, leaf, start, end) -> {
+            final int copied = end - start;
+            for (int i = start; i < end; i++) {
+                results[index + i] = type.getAt(leaf, i);
```

Comment:
> and I think this will most probably not be the bottleneck of an application. 

For operations that need a lot of mutations (e.g. sort), it's best to dump a persistent collection to an array, do the dew and construct the collection back. This needs optimized from and to array operations, hence this commit :)